enhancements/storage/csi-efs-operator.md

```diff
@@ -0,0 +1,341 @@
+    than using `SharedVolume`.
+
+### AWS EFS CSI driver
+
+AWS EFS CSI driver is different than the other CSI drivers we ship. Listing the main differences here for completeness.
+
+* Right now, it does not implement dynamic provisioning. As described above, cluster admin must create the volume in
+  AWS and figure out the networking + security manually.
+  * [Long upstream thread about dynamic provisioning](https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/6)
+  * Upstream [will offer some sort of dynamic provisioning in the next version](https://github.com/kubernetes-sigs/aws-efs-csi-driver/pull/274)
+    by creating subdirectories (Access Points) in existing EFS volumes as PVs. AWS has a limit of 120 Access Points per
+    EFS volume, i.e. only 120 PVs can be created for a single StorageClass.
+    [Upstream issue](https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/310).
+  * Proper dynamic provisioning can come in the future, dealing with the networking and security groups is the hardest
+    part there.
+* AWS EFS volumes do not have any size enforcement. They can hold as much data as apps put there and someone will have to
+  pay for the storage. OCP offer standard metrics for amount of data on PVCs, it's up to the cluster admin to create
+  alerts if the consumption gets too high.
+* It uses [efs-utils](https://github.com/aws/efs-utils) to provide encryption in transport, which is not included in
+  RHEL. We will need to package it, either as an image or RPM package.
+
+## Proposal
+High level only, see design details below.
+
+1. Rewrite current aws-efs-operator to use [CSI driver installation functions of library-go](https://github.com/openshift/library-go/tree/master/pkg/operator/csi/csicontrollerset).
+   * This allows us to share well tested code to install CSI drivers and implement bugfixes / new features at a single
+     place.
+   * It will add common features of CSI driver operators which are missing in the current community operator, such as
+     proxy injection and restarting the drivers when cloud credentials change.
+
+2. Update (rewrite) current aws-efs-operator to watch `ClusterCSIDriver` CR named `efs.csi.aws.com`.
+   * `ClusterCSIDriver` CRD is already provided by OCP and is used to install all other CSI drivers that OCP ships.
+     * Every CSI driver operator watches only its CR, with [a defined name](https://github.com/openshift/api/blob/4b79815405ec40f1d72c3a74bae0ae7da543e435/operator/v1/0000_90_cluster_csi_driver_01_config.crd.yaml#L39).
+       They don't act on or modify CRs of other CSI drivers.
+   * Users must explicitly create the CR to install the CSI driver to allow users to uninstall the driver if they don't
+     like it.
+
+3. Keep existing functionality:
+   * `SharedVolume` CRD in `aws-efs.managed.openshift.io/v1alpha1` API group keeps working as it is now.
+   * Update from the community operator to the supported operator should be as seamless as possible.
```

Comment:
`Update from the community operator to the supported operator` means uninstall the community operator and reinstall the supported operator?
Do we support `update from the supported operator to the community operator` too?

Reply:
Given the complexity of managing "upgrades" (or downgrades) and the low usage of the community operator, is it worth considering *only* providing documentation for such moves instead?